frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -123,6 +148,19 @@ private boolean isBranchNameRepeated(String fileContent, String branchName) {
     return lines.stream().filter(line -> line.equals(branchName)).count() > 1;
   }
 
+  private String getParentBranchName(Path macheteFilePath, String branchName) throws BranchLayoutException {
+    val branchLayoutReader = RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutReader.class);
+    val branchLayout = branchLayoutReader.read(macheteFilePath);
+    val branchLayoutEntry = branchLayout.findEntryByName(branchName);
+    assert branchLayoutEntry != null;
+    val parentBranchEntry = branchLayout.findDirectParentBranch(branchLayoutEntry, branchLayoutEntry);
+    if (parentBranchEntry == null) {
+      return "";
```

Comment:
Hmmm can `null` be returned from this method instead? 🤔 or does empty string make more sense here?

Reply:
Empty String makes more sense to me here. This method should return a String - and empty String is a sign, that evaluation went well, but no parent branch was found (branchName is for a root entry). 
If I can avoid returning `null` then I go for it. Less `null`s, less problems.